tests/ui/drop/lint-if-let-rescope-gated.rs

```diff
@@ -0,0 +1,32 @@
+// This test checks that the lint `if_let_rescope` does not action if either
+// - Edition 2024 is not enabled, or
+// - the feature gate `if_let_rescope` is disabled
```

Comment:
Problem [REV7]: as I realize this is a migration lint, this should only check lint does not fire when feature gate is absent. Sorry!

Reply:
Applied